src/WireMock.Net/Server/FluentMockServer.LogEntries.cs

```diff
@@ -39,7 +39,7 @@ public IEnumerable<LogEntry> FindLogEntries([NotNull] params IRequestMatcher[] m
         {
             var results = new Dictionary<LogEntry, RequestMatchResult>();
 
-            foreach (var log in _options.LogEntries)
+            foreach (var log in _options.LogEntries.ToList())

```

Comment:
I think it's even better to just use the property `LogEntries` here (that property is already made safe...)


Also in the `DeleteLogEntry()` method, it's safer to use the `LogEntries` property.

Reply:
> I think it's even better to just use the property LogEntries here (that property is already made safe...)

Adding/Removing/Setting/Clearing is thread safe for the `LogEntries` property (due to https://github.com/WireMock-Net/WireMock.Net/blob/master/src/WireMock.Net/Util/ConcurrentObservableCollection.cs) but enumerating it is not. 

In my case we have thread 1 calling `FindLogEntries` 
and thread 2 calling a mocked http api. 

Since the `foreach` loop above does not hold a lock, the log entry for the call made by thread 2 is added to the `LogEntries` property. This causes the enumerator in the `foreach` loop to throw an exception. 

I hope that explanation makes sense?

Reply:
> Also in the DeleteLogEntry() method, it's safer to use the LogEntries property

Yup in that method `LogEntries` is perfect, since the `Remove` method on `LogEntries` is thread safe.